src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java

```diff
@@ -73,6 +87,79 @@ public void addImageToCustomer(Integer id, String fileName) {
 
     @Override
     public CustomerDTO findCustomerByUserId(Integer id) {
-        return customerMapper.customerToCustomerDTO(customerRepository.findCustomerByUserId(id));
+        return customerMapper.customerToCustomerDTO(customerRepository.getCustomerByUserId(id));
+    }
+
+    @Override
+    public Page<CustomerDTO> getCustomersByStatus(Pageable pageable, CustomerStatus status) {
+        return customerRepository.findByStatus(status,pageable).map(customerMapper::customerToCustomerDTO);
+    }
+
+    @Override
+    public CustomerDTO updateStatus(Integer id, String status) {
+        Customer customer = customerRepository.findById(id).orElseThrow(() -> new NotFoundException("Customer not found"));
+        customer.setStatus(CustomerStatus.valueOf(status));
+        customerRepository.save(customer);
+        return customerMapper.customerToCustomerDTO(customer);
+    }
+
+    @Override
+    public <T> Page<Customer> findAll(Specification<T> approved, int page, int numberOfProvidersOnPage, String sortBy) {
+        return customerRepository.findAll(approved,PageRequest.of(page, numberOfProvidersOnPage, Sort.by(sortBy).descending()));
+    }
+
+    @Override
+    public void addOrRemoveFromFavourites(Integer customerId, Integer providerId) {
+        if (getCustomerById(customerId).getFavourites().isEmpty()) {
```

Comment:
Why are u returning CustomerDTO, and then transforming providerDTOs to providers?
Why not simple getByID?